### PR TITLE
Bug 1131024 - [System] Move private browser trigger into browser.js, handle private-window request

### PR DESCRIPTION
**Pull request by KevinGrandon**
https://bugzilla.mozilla.org/show_bug.cgi?id=1131024

### CODE REVIEW

**Comment by try-server-hook** [comment]
[KevinGrandon Kevin Grandon (KevinGrandon)](https://github.com/KevinGrandon) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8b8fa2fb6b71)

File: apps/system/js/browser_context_menu.js [hunk 1]
```diff
@@ -250,12 +249,7 @@
   BrowserContextMenu.prototype.newWindow = function(manifest, isPrivate) {
     // For private windows we create an empty private app window.
     if (isPrivate) {
-      var privateBrowserUrl = location.origin + '/private_browser.html';
-      var config = new BrowserConfigHelper({url: privateBrowserUrl});
-      config.oop = true;
-      config.isPrivate = true;
-      var newApp = new AppWindow(config);
-      newApp.requestOpen();
+      browser.newPrivateWindow();
```
Comment (alivedise) [inline apps/system/js/browser_context_menu.js]: Dislike coupling browser here. Use service.request() or event dispatching.

File: apps/system/js/rocketbar.js [hunk 2]
```diff
@@ -647,6 +648,9 @@
       }
 
       switch (e.detail.action) {
+        case 'private-window':
+          browser.newPrivateWindow();
```
Comment (alivedise) [inline apps/system/js/rocketbar.js]: Service.request('newPrivateWindow');

File: apps/system/js/browser.js [hunk 3]
```diff
@@ -13,17 +13,43 @@
...
+        case 'url':
+          handleOpenUrl(UrlHelper.getUrlFromInput(data.url), data.isPrivate);
+          break;
+      }
+    },
+
+    /**
+     * Opens a new private window.
+     */
+    newPrivateWindow: function(url) {
```
Comment (alivedise) [inline apps/system/js/browser.js]: None of your calls give the url argument, could we remove it?

File: apps/system/js/browser_context_menu.js (hunk 1 above)
Comment (KevinGrandon) [inline apps/system/js/browser_context_menu.js]: Yup, that sounds better. Switched to using an event for now, will talk to you tomorrow about using service, and when it makes sense to use either. (I'm not currently clear on that)

**Comment by try-server-hook** [comment]
[KevinGrandon Kevin Grandon (KevinGrandon)](https://github.com/KevinGrandon) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=0d8331c2f7e7)

**Pull request merged**